internal/kafka/internal/handlers/validation.go

```diff
@@ -95,18 +95,18 @@ func ValidateCloudProvider(kafkaService *services.KafkaService, kafkaRequest *db
 
 func ValidateKafkaUpdateFields(kafkaUpdateRequest *private.KafkaUpdateRequest) handlers.Validate {
 	return func() *errors.ServiceError {
-		if stringNotSet(&kafkaUpdateRequest.StrimziVersion) &&
-			stringNotSet(&kafkaUpdateRequest.KafkaVersion) &&
-			stringNotSet(&kafkaUpdateRequest.KafkaIbpVersion) &&
-			stringNotSet(&kafkaUpdateRequest.KafkaStorageSize) {
+		if !stringSet(&kafkaUpdateRequest.StrimziVersion) &&
+			!stringSet(&kafkaUpdateRequest.KafkaVersion) &&
+			!stringSet(&kafkaUpdateRequest.KafkaIbpVersion) &&
+			!stringSet(&kafkaUpdateRequest.KafkaStorageSize) {
 			return errors.FieldValidationError("Failed to update Kafka Request. Expecting at least one of the following fields: strimzi_version, kafka_version, kafka_ibp_version or kafka_storage_size to be provided")
 		}
 		return nil
 	}
 }
 
-func stringNotSet(value *string) bool {
-	return value == nil || len(*value) < 1
+func stringSet(value *string) bool {
+	return value != nil && len(*value) > 0
```

Comment:
A string composed of just blanks is considered 'set'? If not, I believe we should trim the string before checking the length

Reply:
fixed